### ISSUE TITLE
Results capture

### ISSUE THREAD

**Issue by kwolfy** [NagRock/ts-mockito#68]
Whether there are plans to realize capture of results. I could help with the implementation. Something like that:
```typescript
class Foo {
  bar(a: number) {
    return a + 1;
  }
}

const foo = new Foo();
const mockedFoo = spy(foo);

foo(11);
captureResults(mockedFoo.bar).last(); // 12

//Or with breaking changes

capture(mockedFoo.bar).last().result; // 12
capture(mockedFoo.bar).last().args; // [11]

```

**Comment by dreef3** [comment]
What about `capture(mockedFoo.bar())` or maybe add `result` as a (non-enumerable) property to the array of arguments that's returned now?

**Comment by kwolfy** [comment]
Yes, I like this option

**Comment by NagRock** [comment]
Hi, @kwolfy can you tell in which case you need capture of result? If you are spying on real object then you can call spy method and get result.

**Comment by dreef3** [comment]
@NagRock, yes but what if it's not a direct call of the method? Say it's somewhere deep in the code.

**Comment by NagRock** [comment]
Can you give an example? You can always call in spec method on spy object.
